### PR TITLE
cmake: cache: Use generic cache configuration

### PR DESCRIPTION
**Pull request by patrickelectric**
Move from single check to multiple cache options,
allowing the user to select any cache system necessary.

It also adds sccache as an option.

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: {{cookiecutter.project_slug}}/cmake/cache.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+set(CACHE_OPTION "ccache" CACHE STRING "Compiler cache to be used.")
+set(CACHE_OPTION_VALUES "ccache" "sccache")
+set_property(CACHE CACHE_OPTION PROPERTY STRINGS ${CACHE_OPTION_VALUES})
+list(FIND CACHE_OPTION_VALUES ${CACHE_OPTION} CACHE_OPTION_INDEX)
+
+if(${CACHE_OPTION_INDEX} EQUAL -1)
+    message(STATUS "Using custom compiler cache system: ${CACHE_OPTION}, valid entries are ${CACHE_OPTION_VALUES}")
```
Comment (tarcisioe) [inline {{cookiecutter.project_slug}}/cmake/cache.cmake]: Maybe `explicitly supported entries are`? The user could technically use any other executable name and hope it works.

File: {{cookiecutter.project_slug}}/cmake/cache.cmake [hunk 2]
```diff
@@ -0,0 +1,16 @@
+set(CACHE_OPTION "ccache" CACHE STRING "Compiler cache to be used.")
```
Comment (tarcisioe) [inline {{cookiecutter.project_slug}}/cmake/cache.cmake:6]: Does this code support not using any cache? I'm not sure it does, or how.

Comment (patrickelectric) [reply {{cookiecutter.project_slug}}/cmake/cache.cmake:6]: Nops, I added back the `ENABLE_CACHE` option.

File: {{cookiecutter.project_slug}}/cmake/cache.cmake (hunk 1 above)
Comment (patrickelectric) [reply {{cookiecutter.project_slug}}/cmake/cache.cmake]: Done.

**Pull request merged**